engine/src/main/scala/cromwell/engine/workflow/lifecycle/materialization/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -218,19 +217,30 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
                                       workflowOptions: WorkflowOptions,
                                       pathBuilders: List[PathBuilder],
                                       engineIoFunctions: EngineIoFunctions): Parse[EngineWorkflowDescriptor] = {
-    val namespaceValidation: Parse[ValidatedWomNamespace] = sourceFiles.workflowType match {
+    def chooseFactory(factories: List[LanguageFactory]): Option[LanguageFactory] = factories.find(_.looksParsable(sourceFiles.workflowSource))
+
+    val factory: ErrorOr[LanguageFactory] = sourceFiles.workflowType match {
       case Some(languageName) if CromwellLanguages.instance.languages.contains(languageName.toUpperCase) =>
         val language = CromwellLanguages.instance.languages(languageName.toUpperCase)
-        val factory: ErrorOr[LanguageFactory] = sourceFiles.workflowTypeVersion match {
+        sourceFiles.workflowTypeVersion match {
           case Some(v) if language.allVersions.contains(v) => language.allVersions(v).valid
           case Some(other) => s"Unknown version '$other' for workflow language '$languageName'".invalidNel
-          case _ => language.allVersions.head._2.valid
+          case _ => chooseFactory(language.allVersions.values.toList).getOrElse(language.default).valid
         }
-        errorOrParse(factory).flatMap(_.validateNamespace(sourceFiles, workflowOptions, importLocalFilesystem, workflowIdForLogging, engineIoFunctions))
-      case Some(other) => fromEither[IO](s"Unknown workflow type: $other".invalidNelCheck[ValidatedWomNamespace])
-      case None => fromEither[IO]("Need a workflow type here !".invalidNelCheck[ValidatedWomNamespace])
+      case Some(other) => s"Unknown workflow type: $other".invalidNel[LanguageFactory]
+      case None =>
+        val allFactories = CromwellLanguages.instance.languages.values.flatMap(_.allVersions.values)
+        chooseFactory(allFactories.toList).getOrElse(CromwellLanguages.instance.default.default).validNel
+    }
+
+    factory foreach { validFactory =>
+      workflowLogger.info(s"Parsing workflow as ${validFactory.languageName} ${validFactory.languageVersionName}")
+      pushLanguageToMetadata(validFactory.languageName, validFactory.languageVersionName)
```

Comment:
true, although personally I think it's reassuring to log "you said this", and "you actually got this" separately. 

It also makes this code, and the code of any metadata reader, less complex since they can always rely on this field existing and not have to do `if (supplied.workflowType.exists) { ...} else { actualWorkflowType ... }`